darts/tests/models/forecasting/test_regression_models.py

```diff
@@ -2093,6 +2094,201 @@ def test_catboost_model_with_eval_set(self, lgb_fit_patch):
         assert lgb_fit_patch.call_args[1]["eval_set"] is not None
         assert lgb_fit_patch.call_args[1]["early_stopping_rounds"] == 2
 
+    def test_fit_predict_with_categorical_features(self):
+        """Test a RegressionModel with categorical features"""
+        target_df = pd.DataFrame(
+            {
+                "date": pd.date_range("20130101", periods=100),
+                "target": np.random.rand(100),
+            }
+        )
+        past_cov_df = pd.DataFrame(
+            {
+                "date": pd.date_range("20130101", periods=100),
+                "past_cov_cat_1": np.random.randint(0, 5, 100),
+                "past_cov_cat_2": np.random.randint(0, 5, 100),
+                "past_cov_regular": np.random.randint(0, 5, 100),
+            }
+        )
+        model = LightGBMModel(lags=1, lags_past_covariates=1, output_chunk_length=1)
+        with does_not_raise():
+            model.fit(
+                series=TimeSeries.from_dataframe(
+                    df=target_df,
+                    time_col="date",
+                    value_cols="target",
+                    static_covariates=pd.DataFrame(
+                        {"static_cat_1": [1], "static_cat_2": [2]}
+                    ),
+                ),
+                past_covariates=TimeSeries.from_dataframe(
+                    df=past_cov_df,
+                    time_col="date",
+                    value_cols=["past_cov_cat_1", "past_cov_cat_2", "past_cov_regular"],
+                ),
+                categorical_past_covariates=["past_cov_cat_1", "past_cov_cat_2"],
+                categorical_static_covariates=["static_cat_1", "static_cat_2"],
+            )
+            model.predict(1)
```

Comment:
Can we somehow test whether the LightGBM model actually received and used these categorical variables?

Reply:
The LightGBMRegressor doesn't seem to keep track of categorical_features it used in one of its attributes. Hence, best I could think of was mocking it's fit call and check that the value of the categorical_feature argument is in line with what we expect. WDYT? (see `test_lgbm_categorical_features_passed_to_fit_correctly`)